R/comments.R

```diff
@@ -101,16 +101,10 @@ github_comment <- function(text, info = NULL, token = settings$comment_token) {
   } else {
     stop("Expected a pull or a commit, but received ci_build_info() = ", format(info), call. = FALSE)
   }
-  response <- httr::POST(
-    "https://api.github.com",
-    path = file.path("repos", info$user, info$repo, api_subdir, "comments"),
-    body = list(body = jsonlite::unbox(text)),
-    query = list(access_token = token),
-    encode = "json"
-  )
-
-  if (httr::status_code(response) >= 300L) {
-    message(httr::http_condition(response, "error", task = httr::content(response, as = "text")))
-  }
+  req <- httr2::request("https://api.github.com")
+  req <- httr2::req_url_path_append(req, file.path("repos", info$user, info$repo, api_subdir, "comments"))
```

Comment:
You can of course use `file.path()` but `c()` also works

```suggestion
  req <- httr2::req_url_path_append(req, c("repos", info$user, info$repo, api_subdir, "comments"))
```

or even without `c()` it works

```
  req <- httr2::req_url_path_append(req, "repos", info$user, info$repo, api_subdir, "comments")
```

Reply:
Awesome! I like the latter since it totally off-loads handling to {httr2}. Unlikely but `c()` could maybe eventually cause some issue by type coercion that's avoided using `...` instead.